File: frontend/src/types/schedule.ts
export interface ScheduleItem {
  startTime: string;
  endTime: string;
  activity: string;
  reason: string | null;
}

export interface ScheduleProposal {
  proposalId: string;
  title: string;
  date: string;
  items: ScheduleItem[];
}

export interface ScheduleProposalActionResponse {
  proposalId: string;
  status: string;
  schedule: ScheduleItem[] | null;
}

/** One row in the sidebar's applied-schedule history. */
export interface AppliedScheduleSummary {
  scheduleId: string;
  number: number;
  title: string;
  date: string;
  createdAt: string;
  totalHours: number;
  itemCount: number;
}

export interface AppliedScheduleItem {
  id: string;
  startTime: string;
  endTime: string;
  activity: string;
  description: string | null;
}

export interface AppliedScheduleDetail {
  scheduleId: string;
  title: string;
  date: string;
  createdAt: string;
  totalHours: number;
  items: AppliedScheduleItem[];
}

export interface UpdateAppliedScheduleItem {
  id: string;
  startTime: string;
  endTime: string;
  activity: string;
  description: string | null;
}

export interface UpdateAppliedSchedule {
  title: string;
  date: string;
  items: UpdateAppliedScheduleItem[];
}

File: backend/TimeTracker.Api/Models/Schedule.cs
namespace TimeTracker.Api.Models;

/// <summary>
/// A concrete, applied schedule block. Only ever created by
/// ScheduleProposalsController after the authenticated user explicitly
/// approves a pending ScheduleProposal - never written to directly by the agent.
/// Title/Description are copied from the proposal at approval time and then
/// become independently editable - once applied, a Schedule is its own record
/// and no longer tracks the proposal for future changes (ProposalId remains
/// only as an audit trail of what it was created from, and is also the stable
/// grouping key for the several Schedule rows that make up one applied plan).
/// </summary>
public class Schedule
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string Title { get; set; } = string.Empty;
    public DateOnly Date { get; set; }
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }
    public string Activity { get; set; } = string.Empty;
    public string? Description { get; set; }
    public Guid? ProposalId { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
}
